### ISSUE TITLE
Drowsiness detection using EEG signals using DL

### ISSUE THREAD

**Issue by KamakshiOjha** [abhisheks008/DL-Simplified#784]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Drowsiness Detection Using EEG Signals</br>
:red_circle: **Aim** : To develop a deep learning model to detect drowsiness from EEG signals using various algorithms and compare their performance to identify the best-fitted algorithm based on accuracy scores.</br>
:red_circle: **Dataset** : https://figshare.com/articles/dataset/EEG_driver_drowsiness_dataset/14273687</br>
:red_circle: **Approach** : 

1. **Exploratory Data Analysis (EDA)**:
   - Analyze and preprocess the EEG dataset.
   - Visualize EEG signals and class distributions.

2. **Model Development**:
   - Implement the provided deep learning model using TensorFlow and Keras.
   - Develop and implement three additional models:
     1. Convolutional Neural Network (CNN).

3. **Model Training and Evaluation**:
   - Train each model on the EEG dataset.
   - Evaluate models using accuracy, precision, recall, and F1-score.
   - Compare performances to identify the best model.

4. **Visualization and Conclusion**:
   - Visualize training and validation metrics.
   - Provide insights and conclusions based on model comparisons.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Kamakshi Ojha
- GitHub Profile Link : https://github.com/KamakshiOjha
- Email ID : kkamakshiojha@gmail.com
- What is your participant role? (Mention the Open Source program) GSSOC

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
What are the CNN architectures you are planning to use here?
Apart from CNN what are the models you are planning to implement for this EEG dataset?

@KamakshiOjha

**Comment by KamakshiOjha** [comment]
### Input Layer:
- **Input Shape:** (30, 128, 1)
- 
### Convolutional Layers:
- **Branch 1:**
  - `Conv2D` with 16 filters, kernel size (1, 3), ReLU activation
  - `Conv2D` with 32 filters, kernel size (1, 3), ReLU activation
  - `Conv2D` with 64 filters, kernel size (1, 3), ReLU activation
  - `cbam_block` applied to the output

- **Branch 2:**
  - `Conv2D` with 16 filters, kernel size (1, 5), ReLU activation
  - `Conv2D` with 32 filters, kernel size (1, 5), ReLU activation
  - `Conv2D` with 64 filters, kernel size (1, 5), ReLU activation
  - `cbam_block` applied to the output

- **Branch 3:**
  - `Conv2D` with 16 filters, kernel size (1, 7), ReLU activation
  - `Conv2D` with 32 filters, kernel size (1, 7), ReLU activation
  - `Conv2D` with 64 filters, kernel size (1, 7), ReLU activation
  - `cbam_block` applied to the output

### Combined Branch:
- The outputs from the branches are combined using an `Add` layer.
- **Further Convolutional Layers:**
  - `Conv2D` with 64 filters, kernel size (1, 3), ReLU activation
  - `Conv2D` with 128 filters, kernel size (1, 3), ReLU activation
  - `Conv2D` with 128 filters, kernel size (1, 3), ReLU activation

### Global Pooling and Fully Connected Layers:
- `GlobalAveragePooling2D` applied to the last convolutional layer.
- Fully connected (Dense) layers:
  - `Dense` with 512 units, ELU activation
  - `Dense` with 256 units, ELU activation
  - `Dense` with 128 units, ELU activation
  - `Dense` with 32 units, ELU activation
  - Output `Dense` with 2 units, Softmax activation (for classification)

**Comment by KamakshiOjha** [comment]
Since my dataset consists of EEG signals, I've experimented with various CNN architectures to find the best results. Additionally, I've incorporated an attention mechanism into my model to enhance its performance.

**Comment by abhisheks008** [comment]
> ### Input Layer:
> * **Input Shape:** (30, 128, 1)
> 
> ### Convolutional Layers:
> * **Branch 1:**
>   
>   * `Conv2D` with 16 filters, kernel size (1, 3), ReLU activation
>   * `Conv2D` with 32 filters, kernel size (1, 3), ReLU activation
>   * `Conv2D` with 64 filters, kernel size (1, 3), ReLU activation
>   * `cbam_block` applied to the output
> * **Branch 2:**
>   
>   * `Conv2D` with 16 filters, kernel size (1, 5), ReLU activation
>   * `Conv2D` with 32 filters, kernel size (1, 5), ReLU activation
>   * `Conv2D` with 64 filters, kernel size (1, 5), ReLU activation
>   * `cbam_block` applied to the output
> * **Branch 3:**
>   
>   * `Conv2D` with 16 filters, kernel size (1, 7), ReLU activation
>   * `Conv2D` with 32 filters, kernel size (1, 7), ReLU activation
>   * `Conv2D` with 64 filters, kernel size (1, 7), ReLU activation
>   * `cbam_block` applied to the output
> 
> ### Combined Branch:
> * The outputs from the branches are combined using an `Add` layer.
> * **Further Convolutional Layers:**
>   
>   * `Conv2D` with 64 filters, kernel size (1, 3), ReLU activation
>   * `Conv2D` with 128 filters, kernel size (1, 3), ReLU activation
>   * `Conv2D` with 128 filters, kernel size (1, 3), ReLU activation
> 
> ### Global Pooling and Fully Connected Layers:
> * `GlobalAveragePooling2D` applied to the last convolutional layer.
> * Fully connected (Dense) layers:
>   
>   * `Dense` with 512 units, ELU activation
>   * `Dense` with 256 units, ELU activation
>   * `Dense` with 128 units, ELU activation
>   * `Dense` with 32 units, ELU activation
>   * Output `Dense` with 2 units, Softmax activation (for classification)

Cool. Go ahead with this approach.

Assigned @KamakshiOjha
